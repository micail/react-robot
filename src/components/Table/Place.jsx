import React from 'react';
import PropTypes from 'prop-types';

const Place = ({ x, y }) => {
  return (
    <div className="place">
      <p>
        {x}
        ,
        {y}
      </p>
    </div>
  );
};

export default Place;

Place.propTypes = {
  y: PropTypes.number,
  x: PropTypes.number,
};

Place.defaultProps = {
  x: null,
  y: null,
};
