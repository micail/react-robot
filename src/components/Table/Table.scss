.grid-container {
  display: grid;
  grid-template-columns: 5rem 5rem 5rem 5rem 5rem;
  grid-template-rows: 5rem 5rem 5rem 5rem 5rem;
  grid-gap: 1px;
  grid-auto-flow: row;
  align-items: center;
  justify-content: center;
}

.place {
  margin: 0;
  height: 100%;
  width: 100%;
  border: 1px solid #ccc;

  p {
    margin: 0;
    padding: 0;
  }
}