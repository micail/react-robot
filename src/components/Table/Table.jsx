import React from 'react';
import PropTypes from 'prop-types';

import Place from '../Place/Place';
import './Table.scss';


const Table = ({ matrix, position }) => (
  <div className="grid-container">
    {(matrix.map((row, x) => row.map((y) => <Place key={`${x}-${y}`} x={x} y={y} position={position} />))).reverse()}
  </div>
);

export default Table;

Table.propTypes = {
  matrix: PropTypes.arrayOf(Array),
};

Table.defaultProps = {
  matrix: [],
};
