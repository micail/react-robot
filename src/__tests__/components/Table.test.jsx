import React from 'react';
import { shallow, mount } from 'enzyme';

import Table from '../../components/Table/Table';

describe('Table component', () => {
  it('should render table component', () => {
    const component = shallow(<Table />);
    expect(component).toMatchSnapshot();
  });
  it('should render 25 places in the table', () => {
    const matrix = [
      [0, 1, 2, 3, 4], [0, 1, 2, 3, 4],
      [0, 1, 2, 3, 4], [0, 1, 2, 3, 4],
      [0, 1, 2, 3, 4],
    ];
    const flatten = matrix.reduce((sum, a) => sum + a.length, 0);
    const component = mount(<Table matrix={matrix} />);
    expect(component.find('.place').length).toEqual(flatten);
  });
});
