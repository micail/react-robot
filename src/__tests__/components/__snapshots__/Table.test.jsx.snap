// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table component should render table component 1`] = `
<div
  className="grid-container"
/>
`;
